osu.Game/Overlays/Mods/ModSelectOverlay.cs

```diff
@@ -953,7 +957,7 @@ protected override bool OnClick(ClickEvent e)
                     RequestScroll?.Invoke(this);
 
                 // Killing focus is done here because it's the only feasible place on ModSelectOverlay you can click on without triggering any action.
-                Scheduler.Add(() => GetContainingInputManager().ChangeFocus(null));
+                modSelectOverlayInstance.setTextBoxFocus(false);
```

Comment:
This approach is insufficient. It only fixes the bug in the case wherein clicking the columns was _specifically_ what took focus away from the textbox.

If any other component, such as the chat overlay, happens to take away focus, this stops working:

https://github.com/ppy/osu/assets/20418176/caca26da-6b4c-4a17-a9fb-a2f4d7419924

Note how after opening chat, it takes two tab presses to focus the textbox again.

The actual fix will probably involve looking at the actual focus state of the textbox rather than doing tracking in a separate flag.